patches/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java.patch

```diff
@@ -0,0 +1,21 @@
+--- a/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java
++++ b/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java
+@@ -48,6 +_,8 @@
+         GameEvent.RESONATE_14,
+         GameEvent.RESONATE_15
+     };
++    /** @deprecated Neo: use the {@link net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps#VIBRATION_FREQUENCIES data map} instead. */
++    @Deprecated
+     ToIntFunction<GameEvent> VIBRATION_FREQUENCY_FOR_EVENT = Util.make(new Object2IntOpenHashMap<>(), p_297981_ -> {
+         p_297981_.defaultReturnValue(0);
+         p_297981_.put(GameEvent.STEP, 1);
+@@ -101,7 +_,8 @@
+     VibrationSystem.User getVibrationUser();
+ 
+     static int getGameEventFrequency(GameEvent p_281355_) {
+-        return VIBRATION_FREQUENCY_FOR_EVENT.applyAsInt(p_281355_);
++        var data = p_281355_.builtInRegistryHolder().getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.VIBRATION_FREQUENCIES);
++        return data == null ? 0 : data.frequency();
```

Comment:
Instead of removing zero shouldn't this fallback to the vanilla functionality of querying `VIBRATION_FREQUENCY_FOR_EVENT`? At least until 1.20.5

Reply:
For someone to add to the vanilla map they'd need to cast (whicy by itself is an unsafe operation as other mods could change the fields underlying type, or even we could in a patch version) and synchronize (that's if you realise that the type restriction is stupid) (this is an interface, a mixin isn't possible). Any type of fallback, even if for a version, also removes the ability of removing the default entries.

Reply:
What we need to consider is are people in the short span more likely to want to remove default entries or is it possible there are some mods that are adding to them. While yes I haven't released a build of Mekanism yet for 1.20.4, and I also tend to be quick on the uptake of new APIs provided by Neo, it still is an example that some mods might be doing so (given I am)

Also you have do it with a cast (I instance check it), but given it is in an interface it is already static final so modders don't really have a great way of changing the type of it.